#4 Adding Index Structure and Getting Started section (#6)
Inbibo/shift_docs: docs/intro/terminology.md

# Terminology

Here are some of the basic terminology used in Shift:

## Operator

An operator is an execution block that is delegated of performing a specific action given a set of input data if required. From a programming perspective an operator can be seen as a function that takes some input data and processes it to produce some output data. Operators in Shift can be connected to propagate information between each other.  

*\*operator* and *node* will be used interchangeably.  

## Plug

> A plug is a Shift object containing the data that is passed to or from a certain operator. An operator could have input plugs on its left-side and output on its right-side.

Reviewer: ```suggestion
A plug is a Shift object containing the data that is passed to or from a certain operator. An operator has input plugs on its left-side and output on its right-side.
```

Author: Done!
